### DISCUSSION TITLE
Commit with message <Notes added by 'git notes add'> created on release

### DISCUSSION THREAD

**Discussion by nirougi** [semantic-release/semantic-release#2017] [Configuration] [answered]
When using semantic release with npm and bitbucket, it creates a second commit that adds the git notes with the channel information and has this commit message: `Notes added by 'git notes add'`.

Like this:
![image](https://user-images.githubusercontent.com/1561463/124319329-6dbe8080-db50-11eb-9174-e28559fa9d55.png)

Is there a way to prevent that commit having everything in only one, or at least be able to modify the commit message?
In my company we have a strict commit message format that should be followed by that one as well.

This is my config:
```javascript
module.exports = {
  branches: ["master"],
  preset: "angular",
  plugins: [
    [
      "@semantic-release/commit-analyzer",
      { parserOpts: { headerPattern: customCommitMessagePattern } },
    ],
    "@semantic-release/release-notes-generator",
    "@semantic-release/changelog",
    "@semantic-release/npm",
    "@semantic-release/git",
  ],
};
```

I also tried leaving only the `commit-analyzer` and `npm` plugins, but the commit is created as well.

**Comment by gr2m** [comment]
The commit notes are a core feature built into `semantic-release` itself, it's how we keep track of releases across different branches. As `git` is the only common denominator, `git notes` was the only thing suitable for this task. 

However I don't think semantic-release ever creates a commit with the message

> `Notes added by 'git notes add'`.

Might this be something that the git host does where you push your git commits? I've never seen that before

**Reply by nirougi** [reply]
I came to the conclusion that it's Bitbucket being weird.
As far as I understand git notes are commits in ghost branches and Bitbucket's UI shows them as part of the master branch, despite not being related at all.

Those commits are not shown running `git log` locally on master, but they do appear when using bitbucket's API and that was messing things up with a script we have to check commit messages.

Thanks for your help!

**Comment by nirougi** [comment, marked as answer]
It is not a problem with `semantic-releas`, it's actually Bitbucket showing commits that don't belong to the main branch.

**Reply by y0zg** [reply]
@nirougi How it's possible to disable such behaviour? Asked the same question in bitbucket community https://community.atlassian.com/t5/Bitbucket-questions/disable-quot-git-notes-add-quot-behaviour-for-semantic-release/qaq-p/1837322#M66665

**Reply by nirougi** [reply]
@y0zg  I didn't found a way to do it, sorry.
